src/lib.rs

```diff
@@ -1,16 +1,39 @@
+use core::mem;
+use libc;
 use std::ffi::CString;
 use std::io::{Read, Result, Write};
-pub use std::net::{Shutdown, SocketAddr, ToSocketAddrs, IpAddr, Ipv4Addr};
+pub use std::net::{IpAddr, Ipv4Addr, Shutdown, SocketAddr, ToSocketAddrs};
+
+#[repr(C)]
+pub struct IovecRead {
+    pub buf: *mut libc::c_uchar,
+    pub size: usize,
+}
+pub struct IovecWrite {
```

Comment:
Not sure how to write c struct with read/write pointers in rust

Reply:
Like this? https://stackoverflow.com/questions/62126501/how-to-pass-a-c-struct-to-rust